morph/morph.go

```diff
@@ -1,12 +1,23 @@
 package morph
 
 import (
+	"fmt"
 	"math/big"
 	"strconv"
 
 	"github.com/hashicorp/terraform-plugin-go/tftypes"
 )
 
+// ValidateNewValue runs tftypes.ValidateValue before running tftypes.NewValue to ensure the given value will
+// not cause a panic when converted into the specified Type.
+func ValidateNewValue(t tftypes.Type, val interface{}) (tftypes.Value, error) {
```

Comment:
This looks good, but I would be inclined to call this function something like `newValue()` just because the intent of this is to create a new tftypes value and as a side effect we want to validate it before we do that. I found the code slightly confusing to read below where the function is used.

Reply:
Is there anything I can change to make it less confusing? Or which part was confusing?

Reply:
It was just the name of the function – looks good now!